shared/chat/conversation/input-area/user-mention-hud/mention-hud-container.js

```diff
@@ -44,34 +47,30 @@ const mergeProps = (stateProps, dispatchProps, ownProps) => {
     _loadParticipants: dispatchProps._loadParticipants,
     conversationIDKey: stateProps.conversationIDKey,
     filter: stateProps._filter.toLowerCase(),
+    teamType: stateProps.teamType,
     loading: users.length === 0,
     users,
   }
 }
 
 // TODO fix up the typing of this component
-class AutoLoadMentionHud extends React.Component<any> {
-  componentDidMount() {
-    if (this.props.users.length === 0) {
-      // it can never be 0, we don't have a list of participants cached for the general channel or this channel
-      if (!this.props._generalChannelConversationIDKey) {
-        logger.warn(
-          'Mention HUD: no meta found for general channel, loading participants of current channel.'
-        )
-        this.props._loadParticipants(this.props.conversationIDKey)
-        return
-      }
-      logger.info('Mention HUD: no participants in general channel meta, requesting trusted inbox item.')
-      this.props._loadParticipants(this.props._generalChannelConversationIDKey)
-    }
-  }
-
-  render() {
-    return <MentionHud {...this.props} />
-  }
-}
-
 export default compose(
   connect(mapStateToProps, mapDispatchToProps, mergeProps),
-  setDisplayName('UserMentionHud')
-)(AutoLoadMentionHud)
+  setDisplayName('UserMentionHud'),
+  lifecycle({
```

Comment:
I think keeping `AutoLoadMentionHud` is better, since `lifecycle` is poorly typed. (If anything, would be nice to have a better type on it than `Component<any>`...)